src/posts/components/PostsList/components/PostsList.tsx

```diff
@@ -0,0 +1,162 @@
+import { useQuery } from '@tanstack/react-query';
+import { useSearchParams } from 'react-router-dom';
+
+import useGetBoardId from '@/hooks/useGetBoardId';
+import supabase from '@/lib/supabase';
+import { PostWithUser } from '@/posts/types';
+
+import EmptyState from '../../EmptyState';
+import { PostCard } from './PostCard';
+
+export default function PostsList() {
+  const { boardId } = useGetBoardId();
+  const [searchParams] = useSearchParams();
+  const sortBy = searchParams.get('sortBy') || 'comments_count';
+  const searchQuery = searchParams.get('search') || '';
+
+  const filters = Array.from(searchParams.entries()).reduce(
+    (acc, [key, value]) => {
+      if (!['sortBy', 'search'].includes(key)) {
+        acc[key] = acc[key] ? [...acc[key], value] : [value];
+      }
+      return acc;
+    },
+    {} as Record<string, string[]>
+  );
+
+  const { data, isLoading, refetch } = useQuery({
+    queryKey: ['posts', boardId, sortBy, searchQuery, filters],
+    queryFn: async () => {
+      let query = boardId
+        ? supabase.from('posts_with_users').select('*').eq('board', boardId)
+        : supabase.from('posts_with_users').select('*');
+
+      if (searchQuery) {
+        query = query.ilike('title', `%${searchQuery}%`);
+      }
+
+      Object.entries(filters).forEach(([key, values]) => {
+        if (values.length > 1) {
+          if (key !== 'created_at') {
+            const notValues = values.filter((v) => v.startsWith('not:')).map((v) => v.replace('not:', ''));
+            const regularValues = values.filter((v) => !v.startsWith('not:'));
+            if (regularValues.length > 0) {
+              query = query.in(key, regularValues);
+            }
+
+            if (notValues.length > 0) {
+              query = query.not(key, 'in', `(${notValues})`);
+            }
+          }
+          return;
+        }
+
+        const value = values[0];
+
+        const [operator, actualValue] =
+          value.startsWith('not:') ||
+          value.startsWith('on:') ||
+          value.startsWith('after:') ||
+          value.startsWith('on_or_after:') ||
+          value.startsWith('before:') ||
```

Comment:
it's better to have a service function + a hook to map the param and call the service